### PR TITLE
cAdvisor scrape target for kubernetes v1.7+ 

### PR DESCRIPTION
**Pull request by Miouge1**
Since Kubernetes v1.7 the cAdvisor /metrics is on a different port 4194 and needs an extra Prometheus job.

### CODE REVIEW

**Comment by Miouge1** [comment]
@mgoodness can you have a look at this?

File: stable/prometheus/values.yaml [hunk 1]
```diff
@@ -684,6 +684,43 @@ serverFiles:
...
+
+        relabel_configs:
+          - action: labelmap
+            regex: __meta_kubernetes_node_label_(.+)
+          - target_label: __address__
+            replacement: kubernetes.default.svc:443
+          - source_labels: [__meta_kubernetes_node_name]
+            regex: (.+)
+            target_label: __metrics_path__
+            replacement: /api/v1/nodes/${1}:4194/proxy/metrics
```
Comment (nrmitchi) [inline stable/prometheus/values.yaml:730]: I noticed here that you're using the endpoint for 1.7.0 - 1.7.2.

According to the documentation added in https://github.com/prometheus/prometheus/pull/2918, I believe that the configuration for 1.7.3+ should use `replacement: /api/v1/nodes/${1}/proxy/metrics/cadvisor`, no?

Comment (Miouge1) [reply stable/prometheus/values.yaml:730]: The output from `${1}:4194/proxy/metrics` and `${1}/proxy/metrics/cadvisor` are different here is an example from a 1.7.3 cluster:

## `${1}:4194/proxy/metrics`
```
container_cpu_cfs_periods_total{container_name="REDACTED",id="/kubepods/podc79ea696-d5a2-11e7-b0d7-fa163e5e1340/a6e2c7f17a46e1037f69f16cdd8cad3f970172a2f80d4570076be3c525cf2a7b",image="REDACTED",name="k8s_REDACTED",namespace="default",pod_name="REDACTED"} 2671
```
## `${1}/proxy/metrics/cadvisor`
```
container_cpu_cfs_periods_total{id="/kubepods/podc79ea696-d5a2-11e7-b0d7-fa163e5e1340"} 2706
```

Empirical evidence shows that `${1}:4194/proxy/metrics` has a lot more labels available than `${1}/proxy/metrics/cadvisor`, that's why I went with this `replacement` config.

**Comment by Miouge1** [comment]
Resolving merge conflicts in the last commit.

File: stable/prometheus/values.yaml (hunk 1 above)
Comment (simonswine) [reply stable/prometheus/values.yaml:730]: With kubeadm (here on 1.8) the variant with 4194 fails as it is not enabled anymore. (See PR https://github.com/kubernetes/release/pull/356/files). Furthermore I think it is a bad practice to use an non encrypted port for that. I think the we should go with the `metrics/cadvisor` for the every Kubernetes > 1.7 through kubelet's endpoints

At least on 1.8 I see the full labels and that port and it works just fine with a default kubeadm deployment...

File: stable/prometheus/values.yaml [hunk 2]
```diff
@@ -684,6 +684,43 @@ serverFiles:
             target_label: __metrics_path__
             replacement: /api/v1/nodes/${1}/proxy/metrics
 
+
```
Comment (simonswine) [inline stable/prometheus/values.yaml:695]: Can we add some header comment here, why we need this.

Similar to https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml#L53 and https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml#L88

**Review by simonswine — CHANGES_REQUESTED** [review]
Thank you for that update, had hacked that one or two times already it was about the time it gets done in the upstream helm chart :laughing:

**Pull request merged**